### PR TITLE
OHRM5X-2225: Fix candidate page is throwing error after delete hiring manager

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/client/src/orangehrmRecruitmentPlugin/pages/EditJobVacancy.vue [hunk 1]
```diff
@@ -473,11 +473,13 @@ export default {
...
-          label: `${data.hiringManager.firstName} ${data.hiringManager.middleName} ${data.hiringManager.lastName}`,
-          isPastEmployee: data.hiringManager.terminationId ? true : false,
-        };
+        this.vacancy.hiringManager = data.hiringManager.id
+          ? {
+              id: data.hiringManager.id,
+              label: `${data.hiringManager.firstName} ${data.hiringManager.middleName} ${data.hiringManager.lastName}`,
+              isPastEmployee: data.hiringManager.terminationId ? true : false,
+            }
+          : this.$t('general.deleted');
```
Comment (RajithaKumara) [inline src/client/src/orangehrmRecruitmentPlugin/pages/EditJobVacancy.vue:482]: Attach screenshots for both scenarios. with deleted and with not deleted

File: src/client/src/orangehrmRecruitmentPlugin/pages/ViewCandidatesList.vue [hunk 2]
```diff
@@ -222,12 +222,12 @@ export default {
...
             item.lastName
           }`,
-          manager: item?.vacancy?.hiringManager
+          manager: item?.vacancy?.hiringManager?.id
             ? $tEmpName(item.vacancy.hiringManager, {
                 includeMiddle: true,
                 excludePastEmpTag: false,
               })
-            : '',
+            : $t('general.deleted'),
```
Comment (RajithaKumara) [inline src/client/src/orangehrmRecruitmentPlugin/pages/ViewCandidatesList.vue:230]: is this behavior agreed?

File: src/plugins/orangehrmRecruitmentPlugin/Api/CandidateAllowedActionAPI.php [hunk 3]
```diff
@@ -139,9 +139,12 @@ public function getAll(): EndpointResult
      */
     private function isHiringManager(CandidateVacancy $candidateVacancy): bool
     {
-        $hiringMangerEmpNumber = $candidateVacancy->getVacancy()
-            ->getHiringManager()
-            ->getEmpNumber();
+        $hiringMangerEmpNumber = null;
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/CandidateAllowedActionAPI.php:142]: is this backward compatible API change?

File: src/plugins/orangehrmRecruitmentPlugin/Api/CandidateHistoryAPI.php [hunk 4]
```diff
@@ -69,9 +69,11 @@ public function getAll(): EndpointResult
             ->getCandidateVacancyByCandidateId($this->getCandidateId());
         $rolesToExclude = [];
         if ($candidateVacancy instanceof CandidateVacancy) {
-            $hiringMangerEmpNumber = $candidateVacancy->getVacancy()->getHiringManager()->getEmpNumber();
-            if ($hiringMangerEmpNumber !== $this->getAuthUser()->getEmpNumber()) {
-                $rolesToExclude = ['HiringManager'];
+            if ($candidateVacancy->getVacancy()->getHiringManager() !== null) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/CandidateHistoryAPI.php:72]: same here. backward compatible?

File: src/plugins/orangehrmRecruitmentPlugin/Api/CandidateHistoryAPI.php [hunk 5]
```diff
@@ -151,22 +153,24 @@ public function getOne(): EndpointResult
...
-            $allowedWorkflowItems = $this->getUserRoleManager()->getAllowedActions(
-                WorkflowStateMachine::FLOW_RECRUITMENT,
-                CandidateService::STATUS_MAP[WorkflowStateMachine::RECRUITMENT_APPLICATION_ACTION_ATTACH_VACANCY],
-                $rolesToExclude
-            );
-            $disabled = !in_array(
-                WorkflowStateMachine::RECRUITMENT_APPLICATION_ACTION_SHORTLIST,
-                array_keys($allowedWorkflowItems)
-            );
+        if ($vacancy->getHiringManager()->getEmpNumber() !== null) {
```
Comment (RajithaKumara) [inline src/plugins/orangehrmRecruitmentPlugin/Api/CandidateHistoryAPI.php:156]: seems these API changes are not covered in API testing. Seems API changes are backward compatible, as nothing failed

File: src/lib/config/Config.php [hunk 6]
```diff
@@ -52,7 +52,7 @@ class Config
 
     public const PRODUCT_NAME = 'OrangeHRM OS';
     public const PRODUCT_VERSION = '5.5';
-    public const ORANGEHRM_API_VERSION = '2.4.0';
+    public const ORANGEHRM_API_VERSION = '2.5.0';
```
Comment (RajithaKumara) [inline src/lib/config/Config.php:55]: :ok_hand:

File: src/client/yarn.lock [hunk 7]
```diff
@@ -4285,17 +4285,10 @@ __metadata:
...
-  resolution: "caniuse-lite@npm:1.0.30001436"
-  checksum: 7928ac7d93741a81b3005ca4623b133e7d790828be70b26ee55e4860facc59bc344f4092e20034981070a4714f70814c8be4929be4b22728031784f267f69099
-  languageName: node
-  linkType: hard
-
-"caniuse-lite@npm:^1.0.30001426":
-  version: 1.0.30001441
-  resolution: "caniuse-lite@npm:1.0.30001441"
-  checksum: 0f5aa8f7ea4d165e88e0d1eaa44564c5bfee66641f265a1fd959e74f0a7e6bc0207db6c28e2fb63dc8b2cd23e0e3cee06c4f372de11c93c57ff5ff4207962c3f
+"caniuse-lite@npm:^1.0.0, caniuse-lite@npm:^1.0.30001400, caniuse-lite@npm:^1.0.30001426":
```
Comment (RajithaKumara) [inline src/client/yarn.lock:4288]: is this security patch?

**Review by RajithaKumara — CHANGES_REQUESTED** [review]
Please update chaangelog also

File: src/client/src/orangehrmRecruitmentPlugin/pages/ViewCandidatesList.vue (hunk 2 above)
Comment (ManulMax) [reply src/client/src/orangehrmRecruitmentPlugin/pages/ViewCandidatesList.vue:230]: yes. discussed in daily sync

File: src/client/yarn.lock (hunk 7 above)
Comment (ManulMax) [reply src/client/yarn.lock:4288]: Got warning to update this. Discussed with Chamara and said to update this to new version.

File: src/client/src/orangehrmRecruitmentPlugin/pages/EditJobVacancy.vue (hunk 1 above)
Comment (ManulMax) [reply src/client/src/orangehrmRecruitmentPlugin/pages/EditJobVacancy.vue:482]: Screenshot from 2023-06-28 11-39-20
Screenshot from 2023-06-28 11-38-08

File: CHANGELOG.TXT [hunk 8]
```diff
@@ -1,3 +1,18 @@
...
+        - Allowing OrangeHRM Opensource 5x users to create and manage Claims
+    - Supported Environments
+        - PHP - 7.4 to 8.2
+        - MariaDB - 5.5 to 10.11
+        - MySQL - 5.5 to 8.0
+
+    Bug Fixes
+    ---------
+    - https://github.com/orangehrm/orangehrm/issues/1684
+    - https://github.com/orangehrm/orangehrm/issues/1698
```
Comment (RajithaKumara) [inline CHANGELOG.TXT:14]: TODO:: Add other issues also

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**